tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -614,10 +644,17 @@ private static int elemByteSize(DataType dataType) {
     throw new IllegalArgumentException("DataType " + dataType + " is not supported yet");
   }
 
+  private static void throwExceptionIfNotByteOfByteArrays(Object array) {
```

Comment:
I think there's something amiss with the way this method is being used, at least at [line 667](https://github.com/andrewcmyers/tensorflow/blob/46981fb61c393a1b32b6e4569ac2e3e5c47b240f/tensorflow/java/src/main/java/org/tensorflow/Tensor.java#L667) -- the intent of that codes with the exception of this method appears to be to label a byte[] as a STRING datatype, however how this method is being used will throw an exception before returning the STRING datatype for a non-0-length primitive byte array.

(Also, it would be nice if we used class methods giving component types and array checks instead of seeing whether its toString() value is an arcane shorthand for arrays and bytes.)

Reply:
I realize this isn't even your code change - that it is a merge down due to the code which came in with https://github.com/tensorflow/tensorflow/issues/8531 - so i'll avoid harping on it here. Sorry.

Reply:
@quaeler No problem. Can/should you fire off another build to make sure it all works?